File: src/main/java/com/itsci/mju/maebanjumpen/review/service/ReviewService.java
package com.itsci.mju.maebanjumpen.review.service;

import com.itsci.mju.maebanjumpen.review.dto.ReviewDTO;

import java.util.List;

public interface ReviewService {
    List<ReviewDTO> getAllReviews();
    ReviewDTO getReviewById(int id);
    ReviewDTO saveReview(ReviewDTO reviewDto);
    void deleteReview(int id);
    ReviewDTO getReviewByHireId(int hireId);

    // ✅ แก้ไข: เปลี่ยน Review เป็น ReviewDTO
    ReviewDTO updateReview(int id, ReviewDTO reviewDto); // ⬅️ นี่คือตัวที่ถูกต้อง

    List<ReviewDTO> getReviewsByHireId(int hireId);
}

File: src/main/kotlin/com/itsci/mju/maebanjumpen/review/service/ReviewService.kt
package com.itsci.mju.maebanjumpen.review.service

import com.itsci.mju.maebanjumpen.review.dto.ReviewDTO
import com.itsci.mju.maebanjumpen.review.request.CreateReviewRequest
import com.itsci.mju.maebanjumpen.review.request.UpdateReviewRequest

interface ReviewService {
    fun listAllReviews(): List<ReviewDTO>
    fun getReviewById(id: Long): ReviewDTO?

    fun createReview(request: CreateReviewRequest): ReviewDTO

    fun deleteReview(id: Long)

    fun getReviewByHireId(hireId: Long): ReviewDTO?

    fun updateReview(request: UpdateReviewRequest): ReviewDTO

    fun getReviewsByHireId(hireId: Long): List<ReviewDTO>

    fun getReviewsByHousekeeperId(housekeeperId: Long): List<ReviewDTO>
}

